src/Routing/Duplex.purs

```diff
@@ -106,31 +200,101 @@ many1 :: forall f a b.
 many :: forall f a b.
   Foldable f =>
   Alternative f =>
   RouteDuplex a b ->
   RouteDuplex (f a) (f b)
 many (RouteDuplex enc dec) = RouteDuplex (foldMap enc) (Parser.many dec)
 
+-- | Consumes or prints all the remaining segments.
+-- |
+-- |```purescript
+-- | parse rest "" == Right []
+-- | parse (path "a/b" rest) "a/b/c/d" == Right ["c", "d"]
+-- |
+-- | print rest ["a", "b"] == "a/b"
+-- |```
 rest :: RouteDuplex' (Array String)
 rest = RouteDuplex (foldMap Printer.put) Parser.rest
 
+-- | Set a default value which will be returned when parsing fails.
+-- | Does not influence printing in any way.
+-- |
+-- |```purescript
+-- | parse (default 0 $ int segment) "1" == Right 1
+-- | parse (default 0 $ int segment) "x" == Right 0
+-- |```
 default :: forall a b. b -> RouteDuplex a b -> RouteDuplex a b
 default d (RouteDuplex enc dec) = RouteDuplex enc (Parser.default d dec)
 
+-- | Augments the behavior of given codec by making it return `Nothing` if parsing
+-- | fails, or `Just value` if it succeeds.
+-- |
+-- |```purescript
+-- | parse (optional segment) "a"        == Right (Just "a")
+-- | parse (optional segment) ""         == Right Nothing
+-- | 
+-- | print (optional segment) (Just "a") == "a"
+-- | print (optional segment) Nothing    == ""
+-- |```
 optional :: forall a b. RouteDuplex a b -> RouteDuplex (Maybe a) (Maybe b)
 optional (RouteDuplex enc dec) = RouteDuplex (foldMap enc) (Parser.optional dec)
 
+-- | Build codec for custom type out of printer and parser functions.
+-- |
+-- |```purescript
+-- | data Sort = Asc | Desc
+-- |
+-- | sortToString :: Sort -> String
+-- | sortToString = case _ of
+-- |   Asc -> "asc"
+-- |   Desc -> "desc"
+-- |
+-- | sortFromString :: String -> Either String Sort
+-- | sortFromString = case _ of
+-- |   "asc" -> Right Asc
+-- |   "desc" -> Right Desc
+-- |   val -> Left $ "Not a sort: " <> val
+-- |
+-- | sort :: RouteDuplex' String -> RouteDuplex' Sort
+-- | sort = as sortToString sortFromString
+-- |```
 as :: forall s a b. (a -> s) -> (String -> Either String b) -> RouteDuplex s String -> RouteDuplex a b
 as f g (RouteDuplex enc dec) = RouteDuplex (enc <<< f) (Parser.as identity g dec)
 
+-- | Refines a codec of Strings to Ints.
+-- |
+-- | ```purescript
+-- | parse (int segment) "1"  == Right 1
+-- | parse (int segment) "x"  == Left (Expected "Int" "x")
+-- |
+-- | print (int segment) 1    == "1"
+-- | ```
 int :: RouteDuplex' String -> RouteDuplex' Int
 int = as show Parser.int
 
+-- | Refines a codec of Strings to Booleans, where `true` and `false` are the
+-- | strings `"true"` and `"false"`, and other strings are rejected.
+-- |
+-- | ```purescript
+-- | parse (boolean segment) "true"  == Right true
+-- | parse (boolean segment) "x"     == Left (Expected "Boolean" "x")
+-- |
+-- | print (boolean segment) true    == "true"
+-- | ```
 boolean :: RouteDuplex' String -> RouteDuplex' Boolean
 boolean = as show Parser.boolean
 
+-- | This does nothing (internally it's defined as identity).
```

Comment:
This comment for `string` sounds dumb, but I couldn't think of better description :P

Reply:
This mainly exists to work around https://github.com/natefaubion/purescript-routing-duplex/issues/6